PIA

```diff
@@ -168,56 +22,27 @@ CheckScriptDependencies ()
 {
 
 declare -a ARRAY
-ARRAY=()
-
-	Packages=( "$@" )
-	Package=
-	printf "%b\n"
-	for Package in ${Packages[@]}; do
-		hash $Package 2> /dev/null
-		if (( $? > 0 )); then
-			
-			printf "%b" "${XMARK}  ${Package}  Not Found But Required" "\n"
-			ARRAY=(${ARRAY[@]} ${Package})
-		fi
-				
+	for Package in "${Dependencies[@]}" 
+	 do		
+		if ! command -v "$Package" > /dev/null 2>&1; then
+			if [[ $Package == 'dnsutils' ]] ; then
+				if command -v dig > /dev/null 2>&1; then continue; fi
```

Comment:
Dnsutils can't be ran in bash. Always returns "no command". That's because it isn't a single package, it's a combination of a ton of them. The old way dig installation candidate would not be found. Simply changing the word dig to dnsutils in the array with the old logic would never be able to find command dnsutils And always want to install it. That's why the nested if statement says if it is the DNS package, then check if diggs there